Cookbook.Infrastructur/Persistance/RecipebookRepository.cs

```diff
@@ -0,0 +1,283 @@
+                            new CommandDefinition(SqliteCommandTexts.Create,
+                                                  recipe, cancellationToken: token));
+
+            if (result > 0)
+            {
+                await AttachIngredientsToRecipe(recipe, connection, token);
+                await AttachStepsToRecipe(recipe, connection, token);
+                await AttachTagsToRecipe(recipe, connection, token);
+            }
+
+            transaction.Commit();
+
+            return result > 0;
+        }
+
+        public async Task<IEnumerable<Recipe>> GetAllAsync(CancellationToken token)
+        {
+            using var connection = await dbConnectionFactory.CreateConnectionAsync(token);
+            var rawRecipes = await connection.QueryAsync(
+                            new CommandDefinition(SqliteCommandTexts.GetAll,
+                                                  cancellationToken: token));
+
+            if (rawRecipes is null)
+            {
+                return Enumerable.Empty<Recipe>();
+            }
+
+            var recipes = new List<Recipe>();
+
+            foreach (var rawRecipe in rawRecipes)
+            {
+                var recipe = ParseRecipe(rawRecipe);
+
+                recipes.Add(recipe);
+            }
+
+            return recipes.OrderBy(x => x.Title);
+        }
+
+        public async Task<Recipe?> GetByIdAsync(Guid id, CancellationToken token)
+        {
+            using var connection = await dbConnectionFactory.CreateConnectionAsync(token);
+
+            var rawRecipe = await connection.QuerySingleOrDefaultAsync(
+                            new CommandDefinition(SqliteCommandTexts.GetById, 
+                                                  new { id }, cancellationToken: token));
+
+            if (rawRecipe is null)
+            {
+                return null;
+            }
+
+            return ParseRecipe(rawRecipe);
+        }
+
+        public async Task<Recipe?> GetBySlugAsync(string slug, CancellationToken token)
+        {
+            using var connection = await dbConnectionFactory.CreateConnectionAsync(token);
+
+            var rawRecipe = await connection.QuerySingleOrDefaultAsync(
+                            new CommandDefinition(SqliteCommandTexts.GetBySlug, 
+                                                  new { slug }, cancellationToken: token));
+
+            if (rawRecipe is null)
+            {
+                return null;
+            }
+
+            return ParseRecipe(rawRecipe);
+        }
+
+        public async Task<IEnumerable<string>> GetAllTitlesAsync(CancellationToken token)
+        {
+            using var connection = await dbConnectionFactory.CreateConnectionAsync(token);
+
+            var titles = await connection.QueryAsync<string>(
+                                new CommandDefinition(SqliteCommandTexts.GetAllTitles,
+                                                      cancellationToken: token));
+
+            if (titles is null)
```

Comment:
Nem tudom melyik a szebb de tudsz ilyet is `return titles?.Order() ?? Enumerable.Empty<string>();`

Reply:
I like this better :D